tests/test_operators.py

```diff
@@ -1,11 +1,29 @@
 from __future__ import annotations
 
 from qadence2_expressions import (
+    CZ,
+    H,
+    NOT,
     RX,
+    SWAP,
+    X,
+    Y,
+    Z,
+    value,
     variable,
 )
 
 
+def test_unitary_hermitian_operators() -> None:
+    assert H() * H() == value(1)
```

Comment:
Maybe also check squaring these operators work ?

Reply:
Like `H() ** 2 == 1`?

I think that this should be an isolated discussion. That is because the power has some extra simplifications attached, like `(a + b) ** 2 / (a + b) = a + b`. For operators, this could be valid for Hermitian operators, but not trivially generalised.

Reply:
Indeed, I was only speaking for Hermitian operators.

Reply:
Right. Only for Hermitian operators, `X() ** 2 == 1` and `X() ** 3 == X()`.

This can also help with exponential expansion.

Reply:
Also, for unitary Hermitian operators only, factional powers are also simplified now, so`√X() * √X() == X()`.